src/modules/users-permissions/index.js

```diff
@@ -9,6 +9,60 @@ import routes from 'routes'
 
 import PermissionManager from './PermissionManager'
 
+/**
+ * Permissions system
+ * 
+ * This module provides a role & permission list approach to managing user access in Nxus.
+ * Routes (or other guarded functionality) is associated with a Permission name, and permissions
+ * are assigned to Roles. A User may have multiple roles, and a permission may belong to multiple roles.
+ * 
+ * Permissions and roles can also be scoped to specific model objects, allowing users access to just those objects
+ * they own or have been given a role in managing. 
+ * 
+ * ## Usage
+ * `import {permissions} from 'nxus-users'
+ * 
+ * ### Registering permissions and roles
+ * 
+ * `permissions.register('permission-name', ['Default Role'])
+ * 
+ * ### Guarding routes and handlers
+ * 
+ * `permissions.guard('/my/route', 'permission-name')
+ * 
+ * `router.route('/my/route', permissions.guardHandler(::this._myRoute, 'permission-name'))
+ * 
+ * ### Checking for user permissions in handlers/templates
+ * 
+ * `req.user.permissions.allows('permission-name')
+ * `req.user.permissions.allows('permission-name', object)
+ * 
+ * ### Object-level permissions
+ * 
+ * Object role assignments need a collection object that subclasses ObjectRoleModel and overrides the `object` attribute:
+ * 
+ * ```
+  import {ObjectRoleModel} from 'nxus-users'
+  export default ObjectRoleModel.extend{{
+    identity: 'my-object-roles',
+    attributes: { object: { model: 'my-object'}}
+  }}
+ * ````
+ * 
+ * The permissions should be registered with an extra argument naming this model collection:
```

Comment:
Right, though the last sentence is more like "Then the set of permissions from those roles is checked for containing `my-object-permission`". Checks are always against permissions not roles.